### PR TITLE
Changing SDK to use Arcade SignTool

### PR DESCRIPTION
**Pull request by JohnTortugo**
Renaming on the SDK to make it use the Arcade SignTool.

Also did some refactoring on the SignTool to remove unnecessary using imports.

How I tested this:

1) Change PackageReference in Tools.proj to point to Microsoft.DotNet.SignTool
2) Create a NuGet package for Microsoft.DotNet.SignTool
3) Create a new package for the SDK (updating the version)
4) Change the version of the SDK in global.json
5) Configure NuGet.config to find the packages produced in (2) and (3)
6) Try signing

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props [hunk 1]
```diff
@@ -53,7 +53,7 @@
     <MicrosoftDiaSymReaderPdb2PdbVersion Condition="'$(MicrosoftDiaSymReaderPdb2PdbVersion)' == ''">1.1.0-beta1-62506-02</MicrosoftDiaSymReaderPdb2PdbVersion>
     <RoslynToolsModifyVsixManifestVersion Condition="'$(RoslynToolsModifyVsixManifestVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsModifyVsixManifestVersion>
     <RoslynToolsNuGetRepackVersion Condition="'$(RoslynToolsNuGetRepackVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsNuGetRepackVersion>
-    <RoslynToolsSignToolVersion Condition="'$(RoslynToolsSignToolVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsSignToolVersion>
+    <RoslynToolsSignToolVersion Condition="'$(RoslynToolsSignToolVersion)' == ''">1.0.0-dev</RoslynToolsSignToolVersion>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: >dev [](start = 88, length = 3)

We don't want to reference a developer build, do we?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj [hunk 2]
```diff
@@ -33,7 +33,7 @@
       <SignToolArgs Include='"$(ArtifactsConfigurationDir)\"' />
     </ItemGroup>
     
-    <Exec Command="$(NuGetPackageRoot)roslyntools.signtool\$(RoslynToolsSignToolVersion)\tools\SignTool.exe @(SignToolArgs, ' ')"
+    <Exec Command="$(NuGetPackageRoot)Microsoft.DotNet.SignTool\$(RoslynToolsSignToolVersion)\tools\Microsoft.DotNet.SignTool.exe @(SignToolArgs, ' ')"
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj]: >Microsoft.DotNet.SignTool [](start = 38, length = 25)

Nit: lower-case package name.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props (hunk 1 above)
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: >RoslynToolsSignToolVersion [](start = 5, length = 26)

The property name should also be changed to MicrosoftDotNetSignToolVersion

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj [hunk 3]
```diff
@@ -8,14 +8,15 @@
...
     <!-- Using an explicit nuspec file since NuGet Pack target currently doesn't support including dependencies in tools packages -->
     <IsPackable>true</IsPackable>
-
+    <NuspecFile>SignTool.nuspec</NuspecFile>
+    <NuspecBasePath>$(OutputPath)</NuspecBasePath>
+    
     <Description>Build artifact signing tool</Description>
     <PackageTags>Arcade Build Tool Signing</PackageTags>
     <DevelopmentDependency>true</DevelopmentDependency>
+    <Version>1.0.0-dev</Version>
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: ><Version>1.0.0-dev</Version> [](start = 4, length = 28)

Remove

**Comment by tmat** [comment]
Why do you include namespace clean-up in the same commit as functional changes?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj]: Will fix.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props (hunk 1 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: I left it that way because I was afraid to break any script interacting (i.e., setting properties via command line) with the build. Should I go ahead and change it?

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Will make it 1.0.0

Comment (weshaggard) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Yes please update it. If we break anything then we will address those as well.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj (hunk 2 above)
Comment (weshaggard) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj]: Actually on non-windows (assuming we run this there someday) it will be required to be lowercase.

**Comment by maririos** [comment]
ci.dot.net is down, so it would be good to wait until is back and have the other to checks validate the changes before merging

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props [hunk 4]
```diff
@@ -53,7 +53,7 @@
     <MicrosoftDiaSymReaderPdb2PdbVersion Condition="'$(MicrosoftDiaSymReaderPdb2PdbVersion)' == ''">1.1.0-beta1-62506-02</MicrosoftDiaSymReaderPdb2PdbVersion>
     <RoslynToolsModifyVsixManifestVersion Condition="'$(RoslynToolsModifyVsixManifestVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsModifyVsixManifestVersion>
     <RoslynToolsNuGetRepackVersion Condition="'$(RoslynToolsNuGetRepackVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsNuGetRepackVersion>
-    <RoslynToolsSignToolVersion Condition="'$(RoslynToolsSignToolVersion)' == ''">1.0.0-beta2-63011-06</RoslynToolsSignToolVersion>
+    <MicrosoftDotNetSignToolVersion Condition="'$(MicrosoftDotNetSignToolVersion)' == ''">1.0.0</MicrosoftDotNetSignToolVersion>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: >1.0.0 [](start = 90, length = 5)

Have we published 1.0.0? Should definitely be pre-release.

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Talked offline. Waiting for the Arcade build to be fixed and SignTool to be published so I can get a version number to put here.

**Comment by JohnTortugo** [comment]
@dotnet-bot test this please

**Comment by JohnTortugo** [comment]
@tmat I updated the version, can you please take a look again?

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj [hunk 5]
```diff
@@ -5,17 +5,18 @@
     <TargetFrameworks>net461;netcoreapp2.0</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <LangVersion>7.2</LangVersion>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: Why this change?

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj [hunk 6]
```diff
@@ -5,17 +5,18 @@
...
 
     <!-- Using an explicit nuspec file since NuGet Pack target currently doesn't support including dependencies in tools packages -->
     <IsPackable>true</IsPackable>
-
+    <NuspecFile>SignTool.nuspec</NuspecFile>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: Why are you adding a nuspec again? The logic in `eng\Sdk.targets` will properly include the publish output in the package with the generated nuspec.

Comment (tmat) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: I don't think we should make this package an SDK.

Comment (tmat) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: This is a tools package.

Comment (alexperovich) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: The logic in those targets can still apply to this package, including the publish output in the package and fixing up the References to make the generated nuspec correct. The only thing that is really Sdk specific is the `<PackageType>` part, and that can be overridden if it actually causes issues with this package.

Comment (alexperovich) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: We could even go a step farther and just set `<PackAsTool>true</PackAsTool>` in the project and let the .NET Core global tools stuff build this package. This would make the project file much simpler.

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: > Why are you adding a nuspec again?

I added this by mistake. I'll remove it on next commit.

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj (hunk 5 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: In some cases the .net core task was failing due to missing a dependency (Microsoft.Extension.FilesystemGlobbing). Since this isn't a MSBuild dependency and I didn't want to add custom code to load the DLL I decided to copy it.

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj (hunk 6 above)
Comment (tmat) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: > We could even go a step farther and just set `<PackAsTool>true</PackAsTool>` in the project

What is the benefit of making this a global tool? Eventually we want this to just be an msbuild task package.

Comment (alexperovich) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: It would build the exe into the package properly and make the tool usable as an exe if needed. If we are removing the exe entirely and making this a task then thats a bit less useful. My suggestion is to make the Project file as simple as possible. I don't believe that anything in `eng\Sdk.targets` is incompatible with the way this project is used, and it will publish the binaries into the package correctly.

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj (hunk 5 above)
Comment (tmat) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: I don't understand how this property is related to any dependency.

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj (hunk 6 above)
Comment (tmat) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: If you can refactor `Sdk.targets` to two different targets files - one for creating SDK packages and the other for creating tools packages, then I would agree that using the latter would be better than having a custom nuspec.

File: src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj (hunk 5 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/Microsoft.DotNet.SignTool.csproj]: You're right. That change was in another PR. The inclusion of this was also a mistake - as the nuspec directive. Removed.

**Comment by JohnTortugo** [comment]
This have been completed/merged as part of #303
